src/ast.rs

```diff
@@ -81,6 +81,27 @@ impl Value {
             Value::Regex(_) => Type::Regex,
         }
     }
+
+    pub fn as_string(&self) -> Option<&String> {
+        if let Value::String(s) = self {
+            return Some(s);
+        }
+        None
+    }
+
+    pub fn as_int(&self) -> Option<&i64> {
```

Comment:
For copyable type like `i64`, we'd better doing copy-and-return.

Reply:
done.